docs: Explain recommended versions of python + linter concerns

terriko opened the pull request:
* Related to #2618 
* fixes #2492

This adds an explainer to our contributors docs explaining what versions of python we support, which ones they should probably use to start, and explains that the oldest version may not be supported by all our linters.    Also I updated the version of python used in our virtualenv setup docs to be 3.11 in line with our guidance to use the latest or latest-1. (it was previously 3.9)

Molkree commented on CONTRIBUTING.md:
```diff
@@ -9,6 +9,7 @@ If you've already contributed to other open source projects, contributing to the
   - [Code of Conduct](#code-of-conduct)
   - [Development Environment](#development-environment)
   - [Getting and maintaining a local copy of the source code](#getting-and-maintaining-a-local-copy-of-the-source-code)
+  - [Choosing a version of python](#choosing-a-version-of-python)
```
Just a nit but I would capitalize Python everywhere.
```suggestion
  - [Choosing a version of Python](#choosing-a-version-of-python)
```

Molkree commented on CONTRIBUTING.md:
```diff
@@ -85,6 +86,12 @@ Once you've set up your fork, you will find it useful to set up a git remote for
 
 Replace MYUSERNAME with your own GitHub username.
 
+## Choosing a version of python
+
+CVE Binary Tool supports [any version of python that has ongoing security support](https://endoflife.date/python).  There can be a bit of lag before we enable a new release or fully disable an old one.  If you don't need multiple versions of python for a specific bug, we recommend starting with either the most recent version of python or the one before (e.g. the first or second thing on [the "act ...
+
+Note that some of our [linters](#running-linters) (tools used to help with code quality) may not work on the oldest version of python.  That's the last one on [the "active python releases" list](https://www.python.org/downloads/).  So if you're doing development on the oldest version of python, you should switch to a newer version before you run linters on your code or do a `git commit` for best  ...
```
This last sentence is quite long so moving `for best results` to the beginning might make it easier to grasp on the first reading (or not, I'm not a native speaker myself!)

```suggestion
## Choosing a version of Python

CVE Binary Tool supports [any version of Python that has ongoing security support](https://endoflife.date/python).  There can be a bit of lag before we enable a new release or fully disable an old one.  If you don't need multiple versions of Python for a specific bug, we recommend starting with either the most recent version of Python or the one before (e.g. the first or second thing on [the "Active Python Releases" list](https://www.python.org/downloads/)).

Note that some of our [linters](#running-linters) (tools used to help with code quality) may not work on the oldest version of Python.  That's the last one on [the "Active Python Releases" list](https://www.python.org/downloads/).  So if you're doing development on the oldest version of Python, for best results you should switch to a newer version before you run linters on your code or do a `git commit`.
```

Molkree commented on CONTRIBUTING.md:
````diff
@@ -97,16 +104,16 @@ To install it:
 ```
 
-To make a new venv using python 3.9:
+To make a new venv using python 3.11:
````
```suggestion
To make a new venv using Python 3.11:
```

Molkree reviewed (changes requested):
Just a few formatting nits.

And a few related notes:
1. `mypy` is missing in the list of linters
2. This is somewhat opinionated but why use 3rd party `virtualenv` and not a standard library `venv`?
3. Note that as of now only using isort through pre-commit won't work on Python 3.7. The flake8 change is not accepted yet but with the fix I proposed we will just use older flake8 there (but flake8 through pre-commit will similarly fail).

terriko commented:
> 2\. This is somewhat opinionated but why use 3rd party `virtualenv` and not a standard library `venv`?

You know, that's almost certainly a holdover from when we supported older python that didn't have venv properly integrated.  (Or possibly just my own mental holdover from < python 2.7 days!)  We should probably update those.  I'll open a separate bug for it, though.

terriko commented:
Okay, I've rearranged the sentence and added endoflife to the dictionary allow.  I also updated the branch so tests should actually pass now.

I actually debated about whether it should be python or Python, but looking at the rest of the file it's more often 'python' unless we're talking about the Python Software Foundation, so I'm going to stick with lower case for better consistency.  We could probably open up an issue to normalize our usage across all the docs, but it's not important enough to me to dedicate time to right now (Not with 37 open pull requests...) so maybe when I'm a bit more caught up.

terriko commented:
Whoops, missed this in first pass: I've also added mypy to the list of linters, and opened a separate bug for someone to write more explanatory text on how to run it separately.
